gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/MultiHopsFlowToJobSpecCompiler.java

```diff
@@ -289,12 +311,17 @@ private JobSpec jobSpecGenerator(ServiceNode sourceNode, ServiceNode targetNode,
    * Handle the case when edge is not specified.
    * Always select the first available template.
    */
-  private JobSpec jobSpecGenerator(ServiceNode sourceNode, ServiceNode targetNode, FlowSpec flowSpec) {
+  private JobSpec convertEdgeToJobSpec (ServiceNode sourceNode, ServiceNode targetNode, FlowSpec flowSpec) {
```

Comment:
Can you change the name of this function? I prefer not expose concept of 'edge' in this place. 
Let's say, `JobSpecConverter` ?